omod/src/main/java/org/openmrs/web/controller/user/UserFormController.java

```diff
@@ -125,6 +128,24 @@ public String showForm(@RequestParam(required = false, value = "userId") Integer
 			model.addAttribute("secretQuestion", userService.getSecretQuestion(user));
 		}
 		
+		if (user.getPerson().getId() != null && !Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty()) {
+			model.addAttribute("isProvider", true);
+			
+			//Since identifier can be null, check corresponding identifiers and replace the 'nulls' with a "no identifier"
+			List<Provider> providerList = new ArrayList<Provider>(Context.getProviderService().getProvidersByPerson(user.getPerson()));
```

Comment:
Do you mean that in the JSP you can't iterate over the collection or you do you mean you had no way to iterate over the collection here in the controller?